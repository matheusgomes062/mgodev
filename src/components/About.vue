<script setup lang="ts">
import { computed, ref } from 'vue'
import { about } from '~/portfolio'
const { t } = useI18n()

const { name, role, description, resume, social } = about

let width: number
let height: number
let mouseX: number
let mouseY: number
let mouseLeaveDelay: NodeJS.Timeout

const card = ref()

onMounted(() => {
  width = card.value.offsetWidth
  height = card.value.offsetHeight
})

const mousePX = computed(() => {
  return mouseX / width
})

const mousePY = computed(() => {
  return mouseY / height
})

const cardStyle = computed(() => {
  const rX = mousePX.value * 30
  const rY = mousePY.value * -30
  return {
    transform: `rotateY(${rX}deg) rotateX(${rY}deg)`,
  }
})

const cardBgTransform = computed(() => {
  const tX = mousePX.value * -40
  const tY = mousePY.value * -40
  return {
    transform: `translateX(${tX}px) translateY(${tY}px)`,
  }
})

const dataImage = 'https://images.pexels.com/photos/574071/pexels-photo-574071.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260'

const cardBgImage = computed(() => {
  return {
    backgroundImage: `url(${dataImage})`,
  }
})

const handleMouseMove = (e: any) => {
  mouseX = e.pageX - card.value.offsetLeft - width / 2
  mouseY = e.pageY - card.value.offsetTop - height / 2
}

const handleMouseEnter = () => {
  console.log('teste')
  clearTimeout(mouseLeaveDelay)
}

const handleMouseLeave = () => {
  mouseLeaveDelay = setTimeout(() => {
    mouseX = 0
    mouseY = 0
  }, 1000)
}
</script>

<template lang="pug">
section(class="card-wrap" aria-labelledby="about-title" role="region" @mousemove="handleMouseMove" @mouseenter="handleMouseEnter" @mouseleave="handleMouseLeave" ref="card")
  div(class="card" :style="cardStyle")
    div(class="card-bg" :style="[cardBgTransform, cardBgImage]")
    div(class="card-info")
      h1(v-if="name" class="font-bold" id="about-title") {{ t('about.hi_i_am') }} <br/> {{ name }}

      h2(v-if="role" class="mt-6") {{ t('about.i_am') }} {{ t(role) }}

      p(v-if="description" class="mt-10 text-left flex max-w-660px") {{ t(description) }}

      div(class="mt-6 text-center flex flex-row items-end justify-start")
        a(v-if="resume" :href="t(resume)" class="cursor-pointer mr-2" :aria-label="t('about.resume')")
          span(type="button" class="btn btn--outline").
            {{ t('about.resume') }}
        a(v-if="social.github" aria-label="github" :href="social.github" class="link link--icon text-3xl  mr-2 p-0")
          i-mdi-github
        a(v-if="social.stackoverflow" aria-label="stackoverflow" :href="social.stackoverflow" class="link link--icon text-3xl  mr-2 p-0")
          i-mdi-stack-overflow
        a(v-if="social.linkedin" aria-label="linkedin" :href="social.linkedin" class="link link--icon text-3xl  mr-2 p-0")
          i-mdi-linkedin
        a(v-if="social.devto" aria-label="devto" :href="social.devto" class="link link--icon text-3xl  mr-2 p-0")
          i-mdi-dev-to
        a(v-if="social.medium" aria-label="medium" :href="social.medium" class="link link--icon text-3xl  mr-2 p-0")
          i-mdi-medium
</template>

<style scoped lang="scss">
@import './styles/card-parallax.scss'
</style>
