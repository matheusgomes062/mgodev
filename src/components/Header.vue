<script setup lang="ts">
</script>

<template lang="pug">
header(class="w-11/12 h-24 max-w-1100px mx-auto justify-between flex flex-row items-center")
  h3
    a(href="#") title
  Navbar
</template>
