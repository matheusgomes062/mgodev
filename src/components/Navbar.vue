<script setup lang="ts">
import { isDark, toggleDark } from '~/composables'

const { t, availableLocales, locale } = useI18n()
const toggleLocales = () => {
  // change to some real logic
  const locales = availableLocales
  locale.value = locales[(locales.indexOf(locale.value) + 1) % locales.length]
}
</script>

<template lang="pug">
nav(class="flex items-center")
  ul(class="flex mr-6")
    li(class="ml-6")
      a(class="relative pb-1 cursor-pointer link link--nav p-0") {{t('intro.projects')}}
    li(class="ml-6")
      a(class="relative pb-1 cursor-pointer link link--nav p-0") {{t('intro.skills')}}
    li(class="ml-6")
      a(class="relative pb-1 cursor-pointer link link--nav p-0") {{t('intro.contact')}}
    li(class="ml-6")
      a(class="relative pb-1 cursor-pointer link link--nav p-0 text-xl" @click="toggleDark()")
        i-mdi-white-balance-sunny(v-if="isDark")
        i-ic-round-mode-night(v-else)
    li(class="ml-6" @click="toggleLocales")
      a(class="relative pb-1 cursor-pointer link link--nav text-xl")
        i-mdi-translate
</template>
