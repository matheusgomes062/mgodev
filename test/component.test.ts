import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import About from '../src/components/About.vue'
import Header from '../src/components/Header.vue'
import Navbar from '../src/components/Navbar.vue'
import { i18n, pinia } from './setup/setup'

describe('Header.vue', () => {
  it('should render', () => {
    const wrapper = mount(Header, {
      global: {
        plugins: [i18n, pinia],
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('Navbar.vue', () => {
  it('should render', () => {
    const wrapper = mount(Navbar, {
      global: {
        plugins: [i18n, pinia],
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should change dark theme icon when clicked', async() => {
    const wrapper = mount(Navbar, {
      global: {
        plugins: [i18n, pinia],
      },
    })
    await wrapper.get('[data-test="nav_theme"]').trigger('click')
    expect(wrapper.html()).toContain('data-test="sun"')
  })
})

describe('About.vue', () => {
  it('should render', () => {
    const wrapper = mount(About, {
      global: {
        plugins: [i18n],
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
