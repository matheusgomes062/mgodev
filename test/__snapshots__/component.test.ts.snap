// Vitest Snapshot v1

exports[`About.vue > should render 1`] = `
"<section class=\\"flex flex-col items-center mt-12 text-center\\" aria-labelledby=\\"about-title\\" role=\\"region\\">
  <h1 class=\\"font-bold\\" id=\\"about-title\\">Hi, I am <br> Matheus Gomes</h1>
  <h2 class=\\"mt-6\\">A full-stack developer</h2>
  <p class=\\"mt-10 text-center flex max-w-660px\\">Specialized in front-end with proven technical, organizational and communication skills, working with Agile and designing software solutions. Offering strong Vue skills and experience working with SQL (PostgreSQL), .NET Core, Javascript and more. I recently wrote a book on Markdown about Markdown!</p>
  <div class=\\"mt-6 text-center flex flex-row items-end justify-center\\"><a class=\\" mr-2\\" href=\\"https://app.rezi.ai/s/pjj4bdj2XZTaGtmWhpL2\\" aria-label=\\"Resume\\"><span class=\\"btn btn--outline\\" type=\\"button\\">Resume</span></a><a class=\\"link link--icon text-3xl  mr-2 p-0\\" aria-label=\\"github\\" href=\\"https://github.com/matheusgomes062\\"><svg width=\\"1.2em\\" height=\\"1.2em\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\">
        <path fill=\\"currentColor\\" d=\\"M12 2A10 10 0 0 0 2 12c0 4.42 2.87 8.17 6.84 9.5c.5.08.66-.23.66-.5v-1.69c-2.77.6-3.36-1.34-3.36-1.34c-.46-1.16-1.11-1.47-1.11-1.47c-.91-.62.07-.6.07-.6c1 .07 1.53 1.03 1.53 1.03c.87 1.52 2.34 1.07 2.91.83c.09-.65.35-1.09.63-1.34c-2.22-.25-4.55-1.11-4.55-4.92c0-1.11.38-2 1.03-2.71c-.1-.25-.45-1.29.1-2.64c0 0 .84-.27 2.75 1.02c.79-.22 1.65-.33 2.5-.33c.85 0 1.71.11 2.5.33c1.91-1.29 2.75-1.02 2.75-1.02c.55 1.35.2 2.39.1 2.64c.65.71 1.03 1.6 1.03 2.71c0 3.82-2.34 4.66-4.57 4.91c.36.31.69.92.69 1.85V21c0 .27.16.59.67.5C19.14 20.16 22 16.42 22 12A10 10 0 0 0 12 2Z\\"></path>
      </svg></a><a class=\\"link link--icon text-3xl  mr-2 p-0\\" aria-label=\\"stackoverflow\\" href=\\"https://stackoverflow.com/users/10568034/matheus-gomes\\"><svg width=\\"1.2em\\" height=\\"1.2em\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\">
        <path fill=\\"currentColor\\" d=\\"M17.36 20.2v-5.38h1.79V22H3v-7.18h1.8v5.38h12.56M6.77 14.32l.37-1.76l8.79 1.85l-.37 1.76l-8.79-1.85m1.16-4.21l.76-1.61l8.14 3.78l-.76 1.62l-8.14-3.79m2.26-3.99l1.15-1.38l6.9 5.76l-1.15 1.37l-6.9-5.75m4.45-4.25L20 9.08l-1.44 1.07l-5.36-7.21l1.44-1.07M6.59 18.41v-1.8h8.98v1.8H6.59Z\\"></path>
      </svg></a><a class=\\"link link--icon text-3xl  mr-2 p-0\\" aria-label=\\"linkedin\\" href=\\"https://www.linkedin.com/in/matheus-gomes-dev/\\"><svg width=\\"1.2em\\" height=\\"1.2em\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\">
        <path fill=\\"currentColor\\" d=\\"M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77Z\\"></path>
      </svg></a><a class=\\"link link--icon text-3xl  mr-2 p-0\\" aria-label=\\"devto\\" href=\\"https://dev.to/matheusgomes062\\"><svg width=\\"1.2em\\" height=\\"1.2em\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\">
        <path fill=\\"currentColor\\" d=\\"M7.73 11.93c0 1.72-.02 1.83-.23 2.07c-.19.17-.38.23-.76.23l-.51.01l-.03-2.27l-.02-2.27h.52c.35 0 .6.07.77.21c.24.21.26.25.26 2.02M22 7.5v9c0 1.11-.89 2-2 2H4c-1.11 0-2-.89-2-2v-9c0-1.11.89-2 2-2h16c1.11 0 2 .89 2 2M8.93 11.73c-.03-1.84-.05-1.99-.29-2.39c-.4-.68-.85-.84-2.36-.84H5v7h1.21c1.33 0 1.89-.17 2.29-.71c.41-.53.5-.98.43-3.06m4.19-3.23h-1.48c-1.49 0-1.5 0-1.71.28S9.7 9.21 9.7 12v2.96l.27.27c.25.27.31.27 1.71.27h1.44v-1.19l-1.09-.04l-1.1-.03V12.6l.68-.03l.66-.04v-1.19h-1.39V9.7h2.24V8.5m5.88.06c0-.06-.3-.06-.66-.06l-.68.06l-.59 2.35c-.38 1.48-.62 2.27-.67 2.13c-.08-.27-1.14-4.44-1.14-4.49c0-.05-.31-.05-.68-.05h-.69l.41 1.55c.2.87.59 2.28.81 3.15c.34 1.35.46 1.65.75 1.94c.2.22.45.36.61.36c.33 0 .76-.34.9-.73C17.5 14.5 19 8.69 19 8.56Z\\"></path>
      </svg></a><a class=\\"link link--icon text-3xl  mr-2 p-0\\" aria-label=\\"medium\\" href=\\"https://matheusgomes062.medium.com/\\"><svg width=\\"1.2em\\" height=\\"1.2em\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\">
        <path d=\\"M4.37 7.3a.75.75 0 0 0-.25-.65L2.25 4.4v-.34h5.8l4.48 9.83l3.94-9.83H22v.34l-1.6 1.53c-.13.1-.2.28-.17.45v11.24c-.03.17.04.35.17.45l1.56 1.53v.34h-7.84v-.34l1.61-1.57c.16-.15.16-.2.16-.44V8.5L11.4 19.9h-.6L5.57 8.5v7.64c-.07.32.06.64.29.86l2.1 2.57v.33H2v-.33L4.1 17a.94.94 0 0 0 .27-.86V7.3z\\" fill=\\"currentColor\\"></path>
      </svg></a></div>
</section>"
`;

exports[`Header.vue > should render 1`] = `
"<header class=\\"h-32 max-w-1100px justify-between flex flex-row items-center mx-auto w-95/100\\">
  <h3><a class=\\"link font-bold p-0\\" href=\\"https://github.com/matheusgomes062\\">MG.</a></h3>
  <nav class=\\"center nav\\" role=\\"navigation\\" data-v-41458b80=\\"\\">
    <ul class=\\"nav__list\\" data-v-41458b80=\\"\\">
      <li class=\\"nav__list-item\\" data-v-41458b80=\\"\\"><a class=\\"link link--nav\\" href=\\"#projects\\" data-v-41458b80=\\"\\">Projects</a></li>
      <li class=\\"nav__list-item\\" data-v-41458b80=\\"\\"><a class=\\"link link--nav\\" href=\\"#skills\\" data-v-41458b80=\\"\\">Skills</a></li>
      <li class=\\"nav__list-item\\" data-v-41458b80=\\"\\"><a class=\\"link link--nav\\" href=\\"#contact\\" data-v-41458b80=\\"\\">Contact</a></li>
    </ul><button class=\\"nav__theme btn btn--icon\\" type=\\"button\\" aria-label=\\"Toggle dark mode\\" data-test=\\"nav_theme\\" data-v-41458b80=\\"\\"><svg width=\\"1.2em\\" height=\\"1.2em\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\" data-test=\\"moon\\" data-v-41458b80=\\"\\">
        <path fill=\\"currentColor\\" d=\\"M11.93 2.3c-2.04-.5-4.02-.35-5.77.28c-.72.26-.91 1.22-.31 1.71A9.94 9.94 0 0 1 9.5 12a9.94 9.94 0 0 1-3.65 7.71c-.59.49-.42 1.45.31 1.7c1.04.38 2.17.59 3.34.59c6.05 0 10.85-5.38 9.87-11.6c-.61-3.92-3.59-7.16-7.44-8.1z\\"></path>
      </svg></button><button class=\\"nav__theme btn btn--icon\\" type=\\"button\\" aria-label=\\"Change languages\\" data-v-41458b80=\\"\\"><svg width=\\"1.2em\\" height=\\"1.2em\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\" data-v-41458b80=\\"\\">
        <path fill=\\"currentColor\\" d=\\"m12.87 15.07l-2.54-2.51l.03-.03A17.52 17.52 0 0 0 14.07 6H17V4h-7V2H8v2H1v2h11.17C11.5 7.92 10.44 9.75 9 11.35C8.07 10.32 7.3 9.19 6.69 8h-2c.73 1.63 1.73 3.17 2.98 4.56l-5.09 5.02L4 19l5-5l3.11 3.11l.76-2.04M18.5 10h-2L12 22h2l1.12-3h4.75L21 22h2l-4.5-12m-2.62 7l1.62-4.33L19.12 17h-3.24Z\\"></path>
      </svg></button><button class=\\"btn btn--icon nav__hamburger\\" type=\\"button\\" aria-label=\\"Open or close navbar\\" data-v-41458b80=\\"\\"><svg width=\\"1.2em\\" height=\\"1.2em\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\" data-v-41458b80=\\"\\">
        <path fill=\\"currentColor\\" d=\\"M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z\\"></path>
      </svg></button>
  </nav>
</header>"
`;

exports[`Navbar.vue > should render 1`] = `
"<nav class=\\"center nav\\" role=\\"navigation\\" data-v-41458b80=\\"\\">
  <ul class=\\"nav__list\\" data-v-41458b80=\\"\\">
    <li class=\\"nav__list-item\\" data-v-41458b80=\\"\\"><a class=\\"link link--nav\\" href=\\"#projects\\" data-v-41458b80=\\"\\">Projects</a></li>
    <li class=\\"nav__list-item\\" data-v-41458b80=\\"\\"><a class=\\"link link--nav\\" href=\\"#skills\\" data-v-41458b80=\\"\\">Skills</a></li>
    <li class=\\"nav__list-item\\" data-v-41458b80=\\"\\"><a class=\\"link link--nav\\" href=\\"#contact\\" data-v-41458b80=\\"\\">Contact</a></li>
  </ul><button class=\\"nav__theme btn btn--icon\\" type=\\"button\\" aria-label=\\"Toggle dark mode\\" data-test=\\"nav_theme\\" data-v-41458b80=\\"\\"><svg width=\\"1.2em\\" height=\\"1.2em\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\" data-test=\\"moon\\" data-v-41458b80=\\"\\">
      <path fill=\\"currentColor\\" d=\\"M11.93 2.3c-2.04-.5-4.02-.35-5.77.28c-.72.26-.91 1.22-.31 1.71A9.94 9.94 0 0 1 9.5 12a9.94 9.94 0 0 1-3.65 7.71c-.59.49-.42 1.45.31 1.7c1.04.38 2.17.59 3.34.59c6.05 0 10.85-5.38 9.87-11.6c-.61-3.92-3.59-7.16-7.44-8.1z\\"></path>
    </svg></button><button class=\\"nav__theme btn btn--icon\\" type=\\"button\\" aria-label=\\"Change languages\\" data-v-41458b80=\\"\\"><svg width=\\"1.2em\\" height=\\"1.2em\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\" data-v-41458b80=\\"\\">
      <path fill=\\"currentColor\\" d=\\"m12.87 15.07l-2.54-2.51l.03-.03A17.52 17.52 0 0 0 14.07 6H17V4h-7V2H8v2H1v2h11.17C11.5 7.92 10.44 9.75 9 11.35C8.07 10.32 7.3 9.19 6.69 8h-2c.73 1.63 1.73 3.17 2.98 4.56l-5.09 5.02L4 19l5-5l3.11 3.11l.76-2.04M18.5 10h-2L12 22h2l1.12-3h4.75L21 22h2l-4.5-12m-2.62 7l1.62-4.33L19.12 17h-3.24Z\\"></path>
    </svg></button><button class=\\"btn btn--icon nav__hamburger\\" type=\\"button\\" aria-label=\\"Open or close navbar\\" data-v-41458b80=\\"\\"><svg width=\\"1.2em\\" height=\\"1.2em\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\" data-v-41458b80=\\"\\">
      <path fill=\\"currentColor\\" d=\\"M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z\\"></path>
    </svg></button>
</nav>"
`;
